src/components/anchor-navigation/anchor-navigation.test.tsx

```diff
@@ -0,0 +1,364 @@
+  ref1 = React.createRef();
+  ref2 = React.createRef();
+  ref3 = React.createRef();
+  ref4 = React.createRef();
+  ref5 = React.createRef();
+
+  return render(
+    <AnchorNavigation
+      stickyNavigation={
+        <>
+          <AnchorNavigationItem target={ref1}>First</AnchorNavigationItem>
+          <AnchorNavigationItem target={ref2}>Second</AnchorNavigationItem>
+          <AnchorNavigationItem target={ref3}>Third</AnchorNavigationItem>
+          <AnchorNavigationItem target={ref4}>
+            The slightly longer than expected fourth navigation item
+          </AnchorNavigationItem>
+          <AnchorNavigationItem target={ref5}>Fifth</AnchorNavigationItem>
+        </>
+      }
+      {...props}
+    >
+      <Content
+        tabIndex={0}
+        ref={ref1}
+        title="First section"
+        data-role="section-1"
+      />
+      <AnchorSectionDivider />
+      <Content ref={ref2} title="Second section" data-role="section-2" />
+      <AnchorSectionDivider />
+      <Content
+        ref={ref3}
+        title="Third section"
+        noTextbox
+        data-role="section-3"
+      />
+      <AnchorSectionDivider />
+      <Content ref={ref4} title="Fourth section" data-role="section-4" />
+      <AnchorSectionDivider />
+      <Content ref={ref5} title="Fifth section" data-role="section-5" />
+    </AnchorNavigation>
+  );
+};
+
+beforeEach(() => {
+  jest.useFakeTimers();
+});
+
+afterEach(() => {
+  jest.useRealTimers();
+});
+
+test("has proper data attributes applied to elements", () => {
+  renderComponent({ "data-role": "tags-test" });
+  expect(screen.getByTestId("tags-test")).toHaveAttribute(
+    "data-component",
+    "anchor-navigation"
+  );
+  expect(screen.getByRole("list")).toHaveAttribute(
+    "data-element",
+    "anchor-sticky-navigation"
+  );
+
+  const anchorNavigationLinks = screen.getAllByRole("link");
+  anchorNavigationLinks.forEach((anchor) => {
+    expect(anchor).toHaveAttribute("data-element", "anchor-navigation-item");
+  });
+});
+
+test.each([0, 1, 2, 3, 4])(
+  "when navigation item is clicked, the item is selected and the section container is focused",
+  async (i) => {
+    renderComponent();
+
+    const user = userEvent.setup({ advanceTimers: jest.advanceTimersByTime });
+
+    const anchorNavigationLinks = screen.getAllByRole("link");
+    await user.click(anchorNavigationLinks[i]);
+
+    act(() => {
+      jest.advanceTimersByTime(10);
+    });
+    // toHaveStyle still passes even with incorrect styles when used with CSS variables
+    // (see https://github.com/testing-library/jest-dom/issues/461 - the variables seem to be
+    // treated by js-dom as "invalid values" as explained in the first reply) - so using toHaveStyleRule instead
+    SELECTED_ITEM_STYLES.forEach(([property, value]) => {
+      expect(screen.getAllByRole("listitem")[i]).toHaveStyleRule(
+        property,
+        value,
+        { modifier: "& a" }
+      );
+    });
```

Comment:
Thanks for the suggestion, which I agree with. It ended up this way because I was having a lot of trouble with these style tests and didn't want to have to rewrite a bunch of tests each time I changed it - but yes as it ended up there's no real gain to having this `forEach` rather than just having the 2 explicit assertions each time.